Unable to compile on Windows using `make pre-commit`

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

```bash
git -c credential.helper= -c init.defaultBranch=main clone -c core.symlinks=true --recursive --progress -- https://github.com/<username>/polars.git ./polars
cd polars
rustup toolchain install nightly --component miri
cd py-polars
make test
make pre-commit
```

### Log output

```shell
error: unused import: `mimalloc::MiMalloc`
  --> py-polars\src\lib.rs:48:5
   |
48 | use mimalloc::MiMalloc;
   |     ^^^^^^^^^^^^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(unused_imports)]`

error: unused import: `crate::memory::TracemallocAllocator`
  --> py-polars\src\lib.rs:68:5
   |
68 | use crate::memory::TracemallocAllocator;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: associated function `new` is never used
  --> py-polars\src\memory.rs:47:18
   |
44 | impl<A: GlobalAlloc> TracemallocAllocator<A> {
   | -------------------------------------------- associated function in this implementation
...
47 |     pub const fn new(wrapped_alloc: A) -> Self {
   |                  ^^^
   |
   = note: `-D dead-code` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(dead_code)]`

error: could not compile `py-polars` (lib) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `py-polars` (lib test) due to 3 previous errors
make: *** [Makefile:94: clippy] Error 101
(.venv)
```


### Issue description

I've been following the [Contributing](https://docs.pola.rs/development/contributing/#setting-up-your-local-environment) guide, and encountered this error on `Windows`. 

I'm new to `rust`, but think I've resolved this in #16130.

### Expected behavior

Commands in the Makefile should work when linting `main` without any changes.
I think either I've missed something really obvious, or nobody else is using `Windows`.

### Installed versions

<details>

make requirements-all, make build

</details>

--- a/py-polars/src/lib.rs
+++ b/py-polars/src/lib.rs
@@ -45,6 +45,7 @@ mod utils;
 #[cfg(all(target_family = "unix", not(use_mimalloc), not(default_allocator)))]
 use jemallocator::Jemalloc;
 #[cfg(any(not(target_family = "unix"), use_mimalloc))]
+#[allow(unused_imports)]
 use mimalloc::MiMalloc;
 use pyo3::panic::PanicException;
 use pyo3::prelude::*;
@@ -65,6 +66,7 @@ use crate::functions::PyStringCacheHolder;
 use crate::lazyframe::{PyInProcessQuery, PyLazyFrame};
 use crate::lazygroupby::PyLazyGroupBy;
 #[cfg(debug_assertions)]
+#[allow(unused_imports)]
 use crate::memory::TracemallocAllocator;
 use crate::series::PySeries;
 #[cfg(feature = "sql")]

--- a/py-polars/src/memory.rs
+++ b/py-polars/src/memory.rs
@@ -44,6 +44,7 @@ pub struct TracemallocAllocator<A: GlobalAlloc> {
 impl<A: GlobalAlloc> TracemallocAllocator<A> {
     /// Wrap the allocator such that allocations are registered with
     /// tracemalloc.
+    #[allow(dead_code)]
     pub const fn new(wrapped_alloc: A) -> Self {
         Self { wrapped_alloc }
     }

